### DISCUSSION TITLE
Is Akka still the upstream project?

### DISCUSSION THREAD

**Discussion by sigurd-cp** [apache/pekko#1481] [General]
Hey guys,

We all know that Akka decided to change the license to Business Source License (BSL).

But the license would still allow to use all changes with Apache 2.0 after three years. 

> Akka code licensed as BSL reverts to Apache 2.0 with a Change Date of 3 years, i.e. whenever Akka code is released under the BSL, that version will revert from BSL to the Apache license three years later and, then, Apache licensing terms will apply to Akka.

https://www.lightbend.com/akka/license-faq

> Change Date:          TBD ( 3 years after 2.7.0 release )

https://github.com/akka/akka/blob/v2.7.0/LICENSE

Akka 2.7 was released in autumn 2022, there it will be available in around a year.

for akka 2.8 they wrote it explicitly 

> Change Date:          2026-03-16 

https://github.com/akka/akka/blob/v2.8.0/LICENSE

Three years are a long time but still the changes will be available for open source projects. Is it planned to use akka as the upstream project and of course being compatible as far as possible to make that possible?

**Comment by pjfanning** [comment]
We might consider merging particular PRs but it would be on a case by case basis. Since Pekko uses different package names and config names, each PR would need to go through some changes before it could be merged to Pekko.

It is still not clear what the process of changing the license will entail. Ideally, code commits will be made to Akka 2.7, for example, to change the license headers on the source files in the 2.7 branch and to change the LICENSE file in the root of the repo. If this step does not happen, we would need to get legal advice about how to proceed. I suspect that Lightbend will not explicitly change the LICENSE or the source headers so we will probably be forced to go through the legal advise route. The text in https://github.com/akka/akka/blob/v2.7.0/LICENSE is not necessarily going to be acceptable and there is a non zero chance that the advice will be that the license is not acceptable for use by ASF projects.

**Reply by pjfanning** [reply]
If there is a change that has been made to Akka that you would like to see an equivalent change in Pekko, raise an issue. Don't raise an omnibus issue 'to grab all changes' issue, that will be closed as impractical. Do not link to any post license change Akka PRs but you can provide a link to Akka issues and provide your own text describing the requirement.

This is the best approach as someone might take on the task of doing their own Pekko solution well in advance of the Akka code becoming theoretically available to use.
